Automattic/woocommerce-payments
# Add variable analysis phpcs sniffs

Part of #8436

## Changes proposed in this Pull Request

* Adds https://github.com/sirbrillig/phpcs-variable-analysis/ as a dev dependency via composer.
* Adds the `VariableAnalysis.CodeAnalysis.VariableAnalysis` sniff.
* Adds an exception to the `VariableAnalysis.CodeAnalysis.VariableAnalysis` sniff that ignores unused variables prefixed with `_unused_`.


## Testing instructions


1. Remove the `VariableAnalysis.CodeAnalysis.VariableAnalysis` excludes from `phpcs.xml.dist`.
2. Run `npm run lint:php` and locate a file with an unused variable.
3. Prepend `_unused_` to all the unused variables in that file.
4. Re-run phpcs for just that file: `./vendor/bin/phpcs --standard=phpcs.xml.dist <path/to/file>`.
5. Make sure no issues are reported.
    * Note: you may receive reports of mis-matched variable names in doc comments if you don't also update the doc comments.

-------------------


**Post merge**
